### PR TITLE
[test] Test compiling the overlay swiftinterfaces to binary form

### PR DESCRIPTION
**Pull request by jrose-apple**
Let's not regress!

### CODE REVIEW

**Comment by jrose-apple** [comment]
I really hope this list isn't platform-dependent, or else I have to make my expected output checking cleverer.

@swift-ci Please test

File: validation-test/ModuleInterface/verify_all_overlays.swift [hunk 1]
```diff
@@ -0,0 +1,25 @@
+// RUN: %empty-directory(%t)
+// RUN: for x in %platform-sdk-overlay-dir/*.swiftinterface; do [[ $(basename "$x") = Swift.swiftinterface || $(basename "$x") = simd.swiftinterface ]] && continue; %target-swift-frontend "$x" -emit-module -o %t/$(basename "${x/.*}").swiftmodule -disable-objc-attr-requires-foundation-module -enable-resilience -Fsystem %sdk/System/Library/PrivateFrameworks/ || echo -swift-version 4 $(basename "$x") >> %t/failures.txt; done
```
Comment (harlanhaskins) [inline validation-test/ModuleInterface/verify_all_overlays.swift]: This seems like a lot for a RUN line, but I suppose it's alright.

**Review by harlanhaskins — APPROVED** [review]

File: validation-test/ModuleInterface/verify_all_overlays.swift (hunk 1 above)
Comment (jrose-apple) [reply validation-test/ModuleInterface/verify_all_overlays.swift]: :-( Yeah. I could pull it out into some kind of script, I guess? But it's also in SIL/verify_all_overlays.sil.

**Comment by jrose-apple** [comment]
@swift-ci Please test

**Comment by jrose-apple** [comment]
```
Command Output (stdout):
--
1a2
> CoreGraphics.swiftinterface
5c6
< SwiftLang.swiftinterface
---
> UIKit.swiftinterface
```

I guess I need to make the failures platform-specific after all.

**Comment by jrose-apple** [comment]
This still isn't 100% correct because master has fixes that I haven't built with locally, but at least it allows different platforms to be different.

@swift-ci Please test

File: validation-test/ModuleInterface/verify_all_overlays.swift [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+macosx: Foundation.swiftinterface
+ios: Foundation.swiftinterface
+tvos: Foundation.swiftinterface
+watchos: Foundation.swiftinterface
+
+// Actual errors.
+macosx: ObjectiveC.swiftinterface
+ios: ObjectiveC.swiftinterface
+tvos: ObjectiveC.swiftinterface
+watchos: ObjectiveC.swiftinterface
```
Comment (jrose-apple) [inline validation-test/ModuleInterface/verify_all_overlays.swift]: I think this one was fixed already.

File: validation-test/ModuleInterface/verify_all_overlays.swift [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+// Needs to be built as Swift 3.
+ios: CoreGraphics.swiftinterface
+tvos: CoreGraphics.swiftinterface
+watchos: CoreGraphics.swiftinterface
+
+// Actual errors.
+macosx: Foundation.swiftinterface
+ios: Foundation.swiftinterface
+tvos: Foundation.swiftinterface
+watchos: Foundation.swiftinterface
```
Comment (jrose-apple) [inline validation-test/ModuleInterface/verify_all_overlays.swift]: I think this is just #19472.

File: validation-test/ModuleInterface/verify_all_overlays.swift [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+// The following textual interfaces (in alphabetical order) are known not to
+// work.
+
+// Needs to be built as Swift 4.2.
+macosx: CloudKit.swiftinterface
+ios: CloudKit.swiftinterface
+tvos: CloudKit.swiftinterface
+watchos: CloudKit.swiftinterface
+
+// Needs to be built as Swift 3.
```
Comment (jrose-apple) [inline validation-test/ModuleInterface/verify_all_overlays.swift]: This one's about to change, see #19598.

File: validation-test/ModuleInterface/verify_all_overlays.swift [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+ios: CoreGraphics.swiftinterface
+tvos: CoreGraphics.swiftinterface
+watchos: CoreGraphics.swiftinterface
```
Comment (jrose-apple) [inline validation-test/ModuleInterface/verify_all_overlays.swift]: This one's about to change; see #19598.

**Comment by jrose-apple** [comment]
@swift-ci Please test

**Pull request merged**